projects/plugins/jetpack/_inc/client/mixins/url-search/index.js

```diff
@@ -47,10 +46,10 @@ export default {
 		debug( 'search posts for:', keywords );
 		if ( this.props.search && keywords ) {
 			debug( 'replacing URL: ' + searchURL );
-			page.replace( searchURL );
+			window.location.href = searchURL;
 		} else {
 			debug( 'setting URL: ' + searchURL );
-			page( searchURL );
+			window.location.href = searchURL;
```

Comment:
Whoa, this is definitely better if it's working the same way we expect it to. Unless I'm missing something - I don't remember why we were using `Page` here anyway.
A question though - do we need this `if` still? The only thing I see different is the debug statement, which we can probably make the same too.

Reply:
I have no idea whether the different debug message holds any useful information for anyone. It seems to differentiate between doing a search and clearing a search?

As for the origin, it looks like it got copied into Jetpack in #8208 from automattic/dops-components. That seems to have copied it from Calypso in Automattic/dops-components#93. Calypso actually does use page.js for routing, according to various mentions in its README files (e.g. [here](https://github.com/Automattic/wp-calypso/blob/1276e7ecb60e62b3784098c411718bafae2e6b7c/docs/routing.md#route-definitions)).